src/engraving/rendering/score/lyricslayout.cpp

```diff
@@ -651,24 +654,28 @@ void LyricsLayout::checkCollisionsWithStaffElements(System* system, staff_idx_t
     SkylineLine& staffSkylineNorth = systemStaff->skyline().north();
     SkylineLine& staffSkylineSouth = systemStaff->skyline().south();
 
-    int maxVerseAbove = int(lyricsVersesAbove.size());
-    int maxVerseBelow = int(lyricsVersesBelow.size());
+    int maxVerseAbove = !lyricsVersesAbove.empty() ? lyricsVersesAbove.crbegin()->first : 0;
+    int maxVerseBelow = !lyricsVersesBelow.empty() ? lyricsVersesBelow.crbegin()->first : 0;
 
-    for (int verse = maxVerseAbove - 1; verse >= 0; --verse) {
-        SkylineLine verseSkyline = createSkylineForVerse(verse, false, lyricsVersesAbove, system);
-        double minDistance = -verseSkyline.minDistance(staffSkylineNorth);
-        if (minDistance < lyricsMinDist) {
-            double diff = lyricsMinDist - minDistance;
-            moveThisVerseAndOuterOnes(verse, 0, true, -diff, lyricsVersesAbove);
+    for (int verse = maxVerseAbove; verse >= 0; --verse) {
+        if (lyricsVersesAbove.count(verse) > 0) {
```

Comment:
This makes sense, good catch. I find it more readable to check-and-continue on the opposite case, cause it saves an indentation level for the main part of the function. In other words, I usually prefer
```
if (lyricsVersesAbove.count(verse) == 0) {
  continue;
}
```
but that's just coding style. Good job.